### PR TITLE
Replace 'ERR_EXPLAIN' with 'ERR_FAIL_*_MSG' in "platform", "modules/gdnative", "modules/gdscript" directories.

### PR DESCRIPTION
**Pull request by profan**
Also turns a few cases where ifs were used to avoid calling `ERR_EXPLAIN` into usages of the new macro, it leaves usages of `ERR_EXPLAIN` in two places in `modules/gdscript/gdscript_parser.cpp` and in  `modules/gdscript/gdscript_editor.cpp` as they were (as they were disjointed from any direct error macro usage).

The two places in question:
https://github.com/godotengine/godot/blob/fe3cd5175589c27c3564224b317bbc0177a5a735/modules/gdscript/gdscript_parser.cpp#L5764
https://github.com/godotengine/godot/blob/fe3cd5175589c27c3564224b317bbc0177a5a735/modules/gdscript/gdscript_editor.cpp#L637

Part of #31207.

### CODE REVIEW

**Comment by profan** [comment]
.. accidentally managed to get hpvb in there as code owner when I was going through to fix the punctuation, didn't mean to touch `drivers/`, that bit is gone now :P

File: modules/gdnative/arvr/arvr_interface_gdnative.cpp
Comment (akien-mga) [inline modules/gdnative/arvr/arvr_interface_gdnative.cpp]: Is this error still relevant now @BastiaanOlij?

File: modules/gdnative/nativescript/godot_nativescript.cpp
Comment (akien-mga) [inline modules/gdnative/nativescript/godot_nativescript.cpp]: Typo in documentation.

The exclamation marks in this class are maybe a bit overkill too.

File: modules/gdnative/nativescript/nativescript.cpp
Comment (akien-mga) [inline modules/gdnative/nativescript/nativescript.cpp]: Missing dot.

File: modules/gdnative/nativescript/nativescript.cpp
Comment (akien-mga) [inline modules/gdnative/nativescript/nativescript.cpp]: Dot.

Comment (akien-mga) [reply modules/gdnative/nativescript/nativescript.cpp]: Several other missing dots in that class.

File: modules/gdnative/nativescript/godot_nativescript.cpp
Comment (profan) [reply modules/gdnative/nativescript/godot_nativescript.cpp]: There's a few other places where exclamation marks occur, should those be changed as well?

File: modules/gdnative/pluginscript/pluginscript_script.cpp
Comment (akien-mga) [inline modules/gdnative/pluginscript/pluginscript_script.cpp]: Capitalize "PluginScript", fix `s/you/your/` and remove the French space before "?" :)

File: modules/gdnative/pluginscript/pluginscript_script.cpp
Comment (akien-mga) [inline modules/gdnative/pluginscript/pluginscript_script.cpp]: Capitalize "UTF-8".

File: modules/gdscript/gdscript.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript.cpp]: Adding dot at EOL. I'd remove the ":" after "of type", we already have single quotes around the type name.

File: modules/gdscript/gdscript.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript.cpp]: UTF-8.

File: modules/gdscript/gdscript.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript.cpp]: Same here with the unnecessary colon IMO (but that's veeery nitpicky, and feel free to disagree).

File: modules/gdscript/gdscript.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript.cpp]: Typo, there should be an underscore between "_get_property" and "list".

File: modules/gdscript/gdscript.cpp
Comment (profan) [reply modules/gdscript/gdscript.cpp]: I'll just stick to what's common, so I'll remove it :)

File: modules/gdscript/gdscript_function.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript_function.cpp]: That's a bit cryptic, can we find something more explicit @bojidar-bg?
(Also, fix capitalization.)

File: modules/gdscript/gdscript_tokenizer.cpp
Comment (akien-mga) [inline modules/gdscript/gdscript_tokenizer.cpp]: Let's `s/New/recent/`.

File: modules/gdscript/gdscript_function.cpp
Comment (profan) [reply modules/gdscript/gdscript_function.cpp]: Looking at the code, maybe addressing mode -> unknown addressing mode at least?

File: platform/android/audio_driver_opensl.cpp
Comment (akien-mga) [inline platform/android/audio_driver_opensl.cpp]: Fix capitalization.

File: platform/android/audio_driver_opensl.cpp
Comment (akien-mga) [inline platform/android/audio_driver_opensl.cpp]: Same here.

File: platform/android/export/export.cpp
Comment (akien-mga) [inline platform/android/export/export.cpp]: UTF-8.

File: platform/android/export/export.cpp
Comment (akien-mga) [inline platform/android/export/export.cpp]: Maybe `s/dof/DoF/`?

File: platform/android/os_android.cpp
Comment (akien-mga) [inline platform/android/os_android.cpp]: The message was more correct before, either "Error" should be uncapitalized, or if should be preceded by a dot and not a comma.

Comment (profan) [reply platform/android/os_android.cpp]: Since I think I saw it in other code, I'll go with a comma as it is but then uncapitalized instead.

File: platform/javascript/os_javascript.cpp
Comment (akien-mga) [inline platform/javascript/os_javascript.cpp]: Dot (also next line).

File: platform/javascript/os_javascript.cpp
Comment (akien-mga) [inline platform/javascript/os_javascript.cpp]: Add colon as you did for other occurrences.

File: platform/osx/os_osx.mm
Comment (akien-mga) [inline platform/osx/os_osx.mm]: Same style issue with `, Error`.

File: platform/uwp/os_uwp.cpp
Comment (akien-mga) [inline platform/uwp/os_uwp.cpp]: Ditto.

File: platform/uwp/os_uwp.cpp
Comment (akien-mga) [inline platform/uwp/os_uwp.cpp]: Ditto.

File: platform/windows/os_windows.cpp
Comment (akien-mga) [inline platform/windows/os_windows.cpp]: Ditto.

File: platform/windows/os_windows.cpp
Comment (akien-mga) [inline platform/windows/os_windows.cpp]: Ditto.

File: platform/windows/os_windows.cpp
Comment (akien-mga) [inline platform/windows/os_windows.cpp]: This one and the next use `, error` which is fine. If you change previous occurrences back to `. Error`, might be worth doing the same here.

**Comment by akien-mga** [comment]
> it leaves usages of ERR_EXPLAIN in two places in modules/gdscript/gdscript_parser.cpp and in modules/gdscript/gdscript_editor.cpp as they were (as they were disjointed from any direct error macro usage).

CC @bojidar-bg. Maybe those should be turned to `ERR_PRINT`? I tried to traceback the call stack briefly for these methods, and I don't see any direct usage of an `ERR_FAIL` close to setting these err strings.

File: modules/gdnative/nativescript/godot_nativescript.cpp
Comment (akien-mga) [reply modules/gdnative/nativescript/godot_nativescript.cpp]: I'd say yes, I don't see many cases where it would actually be a good practice.

**Comment by akien-mga** [comment]
Left some style nitpicks, otherwise should be good to merge.

File: modules/gdnative/nativescript/godot_nativescript.cpp
Comment (profan) [reply modules/gdnative/nativescript/godot_nativescript.cpp]: At least for that given file I went through and replaced the instances of exclamation marks in the error messages with simple periods then :)

**Comment by profan** [comment]
Resolved the issues that were commented on, let me know once it's good to merge and i'll rebase and squash :+1:

**Comment by akien-mga** [comment]
Seems good to squash before merge :)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by profan** [comment]
just realised i managed to get broken code into master because the CI doesn't actually build the windows end of platform.. 

https://github.com/godotengine/godot/blob/3ea33c0e455f5e52b73ae5da51d0000e8decef7b/platform/windows/os_windows.cpp#L2744-L2750

preparing another PR to fix my mistake :eyes:

**Pull request merged**
